auditing/timescaledb.go

```diff
@@ -0,0 +1,365 @@
+		ConnMaxLifetime *time.Duration
+		MaxOpenConns    *int
+	}
+
+	timescaleAuditing struct {
+		component string
+		db        *sqlx.DB
+		log       *slog.Logger
+
+		config *TimescaleDbConfig
+	}
+
+	timescaledbRow struct {
+		Timestamp time.Time `db:"timestamp"`
+		Entry     []byte    `db:"entry"`
+	}
+
+	sqlCompOp string
+)
+
+const (
+	equals sqlCompOp = "equals"
+	like   sqlCompOp = "like"
+	phrase sqlCompOp = "phrase"
+)
+
+func NewTimescaleDB(c Config, tc TimescaleDbConfig) (Auditing, error) {
+	if c.Component == "" {
+		component, err := defaultComponent()
+		if err != nil {
+			return nil, err
+		}
+
+		c.Component = component
+	}
+
+	if tc.Port == "" {
+		tc.Port = "5432"
+	}
+	if tc.DB == "" {
+		tc.DB = "postgres"
+	}
+	if tc.User == "" {
+		tc.User = "postgres"
+	}
+	if tc.Retention == "" {
+		tc.Retention = "14 days"
+	}
+	if tc.ChunkInterval == "" {
+		tc.ChunkInterval = "1 days"
+	}
+	if tc.CompressionInterval == "" {
+		tc.CompressionInterval = "7 days"
+	}
+
+	source := fmt.Sprintf("host=%s port=%s user=%s dbname=%s password=%s sslmode=disable", tc.Host, tc.Port, tc.User, tc.DB, tc.Password)
+
+	db, err := sqlx.Connect("postgres", source)
+	if err != nil {
+		return nil, fmt.Errorf("could not connect to datastore: %w", err)
+	}
+
+	a := &timescaleAuditing{
+		component: c.Component,
+		log:       c.Log.WithGroup("auditing"),
+		db:        db,
+		config:    &tc,
+	}
+
+	maxIdleConns := 5
+	if tc.MaxIdleConns != nil {
+		maxIdleConns = *tc.MaxIdleConns
+	}
+	a.db.SetMaxIdleConns(maxIdleConns)
+
+	connMaxLifetime := 2 * time.Minute
+	if tc.ConnMaxLifetime != nil {
+		connMaxLifetime = *tc.ConnMaxLifetime
+	}
+	a.db.SetConnMaxLifetime(connMaxLifetime)
+
+	maxOpenConns := 95
+	if tc.MaxOpenConns != nil {
+		maxOpenConns = *tc.MaxOpenConns
+	}
+	a.db.SetMaxOpenConns(maxOpenConns)
+
+	err = a.initialize()
+	if err != nil {
+		return nil, fmt.Errorf("unable to initialize timescaledb backend: %w", err)
+	}
+
+	a.log.Info("connected to timescaledb backend")
+
+	return a, nil
+}
+
+func (a *timescaleAuditing) initialize() error {
+	type txStatement struct {
+		query string
+		args  []any
+	}
+
+	initialSchema := &migrator.Migration{
+		Name: "Initial database schema",
+		Func: func(tx *sql.Tx) error {
+			for _, stmt := range []txStatement{
+				{
+					query: `CREATE EXTENSION IF NOT EXISTS timescaledb`,
+				},
+				{
+					query: `CREATE EXTENSION IF NOT EXISTS pg_stat_statements`,
+				},
+				{
+					query: `CREATE TABLE IF NOT EXISTS traces (
+						timestamp timestamp NOT NULL,
+						entry jsonb NOT NULL,
+						ts tsvector GENERATED ALWAYS AS (to_tsvector('simple', entry)) STORED
+					)`,
+				},
+				{
+					query: `SELECT create_hypertable('traces', 'timestamp', chunk_time_interval => $1::interval, if_not_exists => TRUE)`,
+					args:  []any{a.config.ChunkInterval},
+				},
+				{
+					query: `ALTER TABLE traces SET (
+						timescaledb.compress,
+						timescaledb.compress_orderby = 'timestamp'
+					)`,
+				},
```

Comment:
Please do not alter the table on every startup, this is always required so put it into the "CREATE TABLE" statement.

Reply:
I think with the [migrator](https://github.com/lopezator/migrator) this should not happen because it keeps track of the applied versions in a separate table.

Reply:
OK, but why doing this in two steps ? Dont see the point and makes it hard to read

Reply:
I took the commands for enabling compression 1 by 1 from the documentation of TimescaleDB: https://docs.timescale.com/use-timescale/latest/compression/compression-policy/#enabling-compression
Also I don't know how to combine a `SET` with a table creation. You need to help me with that if this is possible.

Reply:
OK, i do not want to press here too much, lets keep it as it is.